File: samples/Test/Classes/Control/ScrollViewTest.cpp

#include "ScrollViewTest.h"

ScrollViewTest::ScrollViewTest()
{
    this->setTitle("CAScrollView");
}

ScrollViewTest::~ScrollViewTest()
{
}

void ScrollViewTest::viewDidLoad()
{
    auto scrollView = CAScrollView::createWithLayout(DLayoutFill);
    scrollView->setMinimumZoomScale(0.2f);
    scrollView->setMaximumZoomScale(5.0f);
    scrollView->setMultitouchGesture(CAScrollView::MultitouchGesture::ZoomAndRotate);
    scrollView->setViewSize(DSize(2160, 3840));
    scrollView->setReachBoundaryHandOverToSuperview(false);
    this->getView()->addSubview(scrollView);

    auto imageView = CAImageView::createWithImage(CAImage::create("image/h1.png"));
    imageView->setLayout(DLayoutFill);
    scrollView->addSubview(imageView);

    scrollView->onDidMoved([=]
    {
        CCLog("onDidMoved");
    });
    scrollView->onStopMoved([=]
    {
        CCLog("onStopMoved");
    });
    scrollView->onWillBeginDragging([=]
    {
        CCLog("onWillBeginDragging");
    });
    scrollView->onDragging([=]
    {
        CCLog("onDragging");
    });
    scrollView->onDidEndDragging([=]
    {
        CCLog("onDidEndDragging");
    });
    scrollView->onDidZoom([=]
    {
        CCLog("onDidZoom");
    });
    scrollView->onTouchUpWithoutMoved([=](DPoint point)
    {
        CCLog("onTouchUpWithoutMoved");
    });
}

void ScrollViewTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

File: samples/Test_JS/Resources/js/Control/ScrollViewTest.js
/**
 * Created by zhanglei on 16/8/8.
 */
var ScrollViewTest = ca.CAViewController.extend({

    ctor: function () {
        this._super();
    },

    viewDidLoad: function() {

        this.colorArr = new Array();

        this.scrollView = ca.CAScrollView.createWithLayout(ca.DLayout.fill());
        this.scrollView.setMinimumZoomScale(0.2);
        this.scrollView.setMaximumZoomScale(5.0);
        this.scrollView.setMultitouchGesture(ca.CAScrollView.MultitouchGesture.ZoomAndRotate);
        this.scrollView.setViewSize(ca.DSize.set(2160,3840));
        this.scrollView.setReachBoundaryHandOverToSuperview(false);
        this.getView().addSubview(this.scrollView);

        this.p_imageView = ca.CAImageView.createWithImage(ca.CAImage.create("image/h1.png"));
        this.p_imageView.setLayout(ca.DLayout.fill());
        this.scrollView.addSubview(this.p_imageView);

        this.scrollView.onDidMoved(this.scrollViewDidMoved.bind(this));
        this.scrollView.onStopMoved(this.scrollViewStopMoved.bind(this));
        this.scrollView.onWillBeginDragging(this.scrollViewWillBeginDragging.bind(this));
        this.scrollView.onDragging(this.scrollViewDragging.bind(this));
        this.scrollView.onDidEndDragging(this.scrollViewDidEndDragging.bind(this));
        this.scrollView.onDidZoom(this.scrollViewDidZoom.bind(this));
        this.scrollView.onTouchUpWithoutMoved(this.scrollViewTouchUpWithoutMoved.bind(this));
    },

    scrollViewDidMoved: function (view)
    {
        log("DidMoved-->");
    },

    scrollViewStopMoved: function ( view)
    {
        log("StopMoved-->");
    },

    scrollViewWillBeginDragging: function ( view)
    {
        log("BeginDragging-->");
    },

    scrollViewDragging: function ( view)
    {
        log("BeginDragging-->");
    },

    scrollViewDidEndDragging: function ( view)
    {
        log("DidEndDragging-->");
    },

    scrollViewDidZoom: function ( view)
    {
        log("DidZoom-->");
    },

    scrollViewTouchUpWithoutMoved: function ( view)
    {
        log("DidEndDragging-->");
    }
});
